presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSink.java

```diff
@@ -14,10 +14,11 @@
 package com.facebook.presto.spi;
 
 import com.facebook.presto.spi.block.Block;
+import io.airlift.slice.Slice;
 
 public interface ConnectorPageSink
 {
     void appendPage(Page page, Block sampleWeightBlock);
 
-    String commit();
+    Iterable<Slice> commit();
```

Comment:
Unless we need to support "streaming" collections (i.e., you don't necessarily know the size at the time you return from the method or you need to generate them lazily), it's probably better to return `Collection`, otherwise certain things like java 8 `streams` become cumbersome to use.

A good rule of thumb is to prefer the most general argument types possible and return the most specific types possible, and refer to interfaces instead of concrete classes (e.g., List vs ArrayList, etc)